### PR TITLE
Compile Geant4 with GDML support

### PR DESCRIPTION
**Pull request by sawenzel**
This enables Geant4 to import/export the standard
GDML file format for detector description.
Needed for geometry related developments and improvements
and in the interest of building a generic Geant4 distribution.

### CODE REVIEW

File: xercesc.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0
+# Our environment
+setenv XERCESC_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
```
Comment (ktf) [inline xercesc.sh]: Please use set, not `setenv`.

File: xercesc.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+prepend-path LD_LIBRARY_PATH \$::env(XERCESC_ROOT)/lib
```
Comment (ktf) [inline xercesc.sh]: `prepend-path LD_LIBRARY_PATH \$XERCESC_ROOT/lib`

File: xercesc.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+$([[ ${ARCHITECTURE:0:3} == osx ]] && echo "prepend-path DYLD_LIBRARY_PATH \$::env(XERCESC_ROOT)/lib")
```
Comment (ktf) [inline xercesc.sh]: `DYLD_LIBRARY_PATH` is not used anymore. Please drop this line.

File: xercesc.sh (hunk 1 above)
Comment (sawenzel) [reply xercesc.sh]: ok (but same in arrow.sh)

File: xercesc.sh (hunk 2 above)
Comment (sawenzel) [reply xercesc.sh]: ok (but same in arrow.sh)

**Comment by ktf** [comment]
we need to fix arrow as well. the linter did not spot it because the arrow recipe was not changed since the introduction of the new check.

**Comment by sawenzel** [comment]
I consider this PR ready to be merged if the tests go through. @MichaelLettrich, @ktf  : Can you take care of the merge, please.

**Comment by ktf** [comment]
I will do it after today daily, since we are on friday and I want to have at least one build before the weekend.

**Comment by sawenzel** [comment]
There is a persisting problem which I could not yet solve:
```
cling::DynamicLibraryManager::loadLibrary(): libxerces-c-3.2.so: cannot open shared object file: No such file or directory
Error in <TInterpreter::TCling::AutoLoad>: failure loading library libgeant4vmc.so for TG4RunConfiguration
In file included from input_line_9:1:
/mnt/mesos/sandbox/sandbox/O2/Detectors/gconfig/g4Config.C:61:31: error: unknown type name 'TG4RunConfiguration'
  auto runConfiguration = new TG4RunConfiguration("geomRoot", physicsSetup, "stepLimiter+specialCuts",
                             ^
```
which happens during unit testing. I assume that we are missing some environment variable during the build process. Any idea? I already defined LD_LIBRARY_PATH in the new xercesc module as well as alibuild header.

**Comment by ihrivnac** [comment]
Hi @sawenzel ,
Could you try to disable building shared libraries by adding -DBUILD_SHARED_LIBS=OFF in cmake instructions? The build will then create a static library for which you don't need to update the dynamic path. 
I am using an older xerces-c version (3.1.1) which build the static library by default and I don't set an extra path to it anywhere  when using it with G4/Geant4 VMC stand-alone.

**Comment by sawenzel** [comment]
@ihrivnac : Thanks for hint. Let's see if it goes through.

**Comment by sawenzel** [comment]
@ihrivnac : New problems now (at least on CC7). Someone in the compile chain needs to link against libcurl because libG4persistency tries to access these symbols. I am afraid this probably needs patching CMakeFiles which is not immediately in my hands.
Should we try making the shared lib version work instead?

**Comment by ihrivnac** [comment]
@sawenzel: According to Xerces-c documentation, the option -Dnetwork:BOOL=OFF can switch off this dependency. But when I tried this one on my Mac I got another undefined symbols when building Geant4 with this version of xerces-c, so I think it will be better to go back to shared library and try to resolve the path to the library.

**Comment by sawenzel** [comment]
It should work now. The reason was that xercesc decided to install its libs in `lib64` on CC7 rather than `lib`. This is why it worked on ubuntu.

**Comment by sawenzel** [comment]
The MacOS CI seems stuck. 18 hours and counting.

**Comment by sawenzel** [comment]
This is finally green. @ktf , @pzhristov : Can it be merged?

**Pull request merged**